tests/conftest.py

```diff
@@ -86,3 +88,20 @@ def ids_by_os(read_json_fixture):
             systems_by_version[key].update([system["id"]])
 
     return systems_by_version
+
+
+@pytest.fixture
+def make_systems():
+    def _make_systems(count=3):
+        system_ids = set()
+        systems_detail = set()
+
+        for n in range(0, count):
+            system_id = uuid4()
+            system = SystemInfo(id=system_id, display_name=f"System {n}")
+            system_ids.add(system_id)
+            systems_detail.add(system)
```

Comment:
This could also be written using comprehensions.
```suggestion
        systems_detail = {SystemInfo(id=system_id, display_name=f"System {n}") for n in range(count)}
        system_ids = {system.id for system in systems_detail}
```